tests/test_get_paragraphs.py

```diff
@@ -0,0 +1,270 @@
+        assert metadata_paras == expected_metadata_paras
+        assert "".join(mtdt_p["value"] for mtdt_p in metadata_paras) == TestToyData.TEXTS[2]
+
+    def test_get_paragraph_by_nested_and_merged_divs(self):
+        """test_mark_paragraph_by_nested_divs
+
+        Example:
+            html = (
+                '<html><body><div class="div-level-1">blablabla blablabla blablabla blablabla blablabla blablabla
+                '<div class="div-level-2">tidi tidi tidi tidi</div></div></body></html>'
+            )
+        """
+
+        expected_metadata_paras = [
+            {
+                "char_end_idx": 80,
+                "char_start_idx": 0,
+                "key": "paragraph",
+                "marker": "div",
+                "type": "local",
+                "value": TestToyData.TEXTS[3],
+            },
+        ]
+        metadata_paras = get_paragraphs(TestToyData.METADATA_HTMLS[3], TestToyData.TEXTS[3])
+        assert metadata_paras == expected_metadata_paras
+
+
+class TestRealData:
+    """The test case is from a small portion of `c4-en-html_cc-main-2019-18_pq00-000`[11].
+
+    With the second last item of the output, the test shows the spec of dangling double LF.
+
+    Example:
+        html = (
+            "..."
+            '<div class="container"><div class="sidebars row"><div class="col-xs-12 col-sm-3">'
+            '<div class="sidebar sidebar-1">...<div class="tagcloud">'
+            "<a ...>CITY MAP</a><a ...>CINQUE TERRE</a><a ...>CITY</a><a ...>HOW TO GET</a><a ...>ACCOMODATION</a>"
+            "<a ...>TRANSPORTS AND BOATS</a><a ...>EVENTS</a><a ...>GULF</a><a ...>PORTOVENERE</a>"
+            "<a ...>NATURAL PARKS</a><a ...>LERICI</a><a ...>THE DISTRICT</a><a ...>HIKING</a><a ...>FOOD CULTURE</a>"
+            "<a ...>INFO CENTER</a>"
+            "..."
+            '<div class="sh_contact"> <span ...>Comune della Spezia</span><br>'
+            "<span ...>Piazza Europa, 1 | 19124 La Spezia</span><br> "
+            "<a ...>infocenterlia@comune.sp.it</a><br>"
+            "..."
+            '<div id="copyright" class="container-full">'
+            "<div ...></div>"
+            "<a ...>...</a><br>"  # Dangling double LF begins
+            "<br>"                # Dangling double LF ends
+            "<span ...>Comune della Spezia - Piazza Europa, 1 |&nbsp;19124 La Spezia - P. IVA 00211160114 | "
+            "<a ...>Cookie Policy</a>&nbsp;| <a ...>...</a></p>"
+            "..."
+        )
+    """
+
+    TEXT = (
+        "CITY MAP CINQUE TERRE CITY HOW TO GET ACCOMODATION "
+        "TRANSPORTS AND BOATS EVENTS GULF PORTOVENERE "
+        "NATURAL PARKS LERICI THE DISTRICT HIKING FOOD CULTURE "
+        "INFO CENTER\n"
+        "Comune della Spezia\n"
+        "Piazza Europa, 1 | 19124 La Spezia\n"
+        "infocenterlia@comune.sp.it\n"
+        "\n\n"
+        "Comune della Spezia - Piazza Europa, 1 | 19124 La Spezia - P. IVA "
```

Comment:
A small decision to make for the dangling/standalone "`\n\n`." Do we want to:
1. Let it be;
1. Remove it, and yet it won't be an reverse operation like `join()` <-> `split()`;
2. Attach it to the previous paragraph.

Reply:
@timoschick @SaulLu Sorry that I didn't even notice that I failed to mention you guys for this.

Reply:
@timoschick @norakassner +CC @manandey As the previous comment said, I won't change anything for this PR just yet. Right now I'm working on actually splitting paragraphs from HTMLs. That task uses my free GCP credit and probably no need for a PR here. No technical difficulty so far.
However, I'm still (literally) under the weather and busy with my day job, so my progress isn't as good as I hope. That being said, I am sure that I can get something done this weekend.